zio-http/src/main/scala/zhttp/service/ServerResponseWriter.scala

```diff
@@ -103,6 +103,12 @@ private[zhttp] final class ServerResponseWriter[R](
       case HttpData.JavaFile(unsafeGet) =>
         unsafeWriteFileContent(unsafeGet())
         releaseAndRead(jReq)
+
+      case HttpData.UnsafeAsync(unsafeRun) =>
+        unsafeRun { _ => msg =>
+          ctx.writeAndFlush(msg)
+          if (!msg.isInstanceOf[LastHttpContent]) ctx.read(): Unit
+        }
```

Comment:
This will allow responses to use the HttpData received in the request directly. This is also going significantly more performant than echo using streams.